### PR TITLE
Add support for computing remainder with Decimal128 operands with more precision on Spark 3.4 [databricks]

### LINKED ISSUE #8330: [BUG] Handle Decimal128 computation with overflow of Remainder on Spark 3.4
**Describe the bug**
When computing the Remainder for Decimal types, the existing algorithm can only handle what previous versions of Spark could handle as far as overflow of the Decimal128 operands. This would mean that if an upcast cannot be performed because of limitations of precision, an exception would be thrown (ANSI mode) or the result would be null. Spark 3.4 actually computes this remainder now because it can use the BigDecimal type from JAVA and round from there.  

**Steps/Code to reproduce bug**
Try to compute the remainder between 2 large decimal 128 values:

PySpark example:
```
data = [[Decimal('5776949384953805890688943467625198736'), Decimal('-67337920196996830.354487679299')]]
schema = StructType([
    StructField("a", DecimalType(38,0), True),
    StructField("b", DecimalType(27,7), True)])
df = spark.createDataFrame(data=data, schema=schema)

out = df.selectExpr("a", "b", r"a % b").collect()
```

**Expected behavior**
When running in Spark 3.4, this should return the correct value for `a % b` (in other versions of Spark, it returns null)

### PR DESCRIPTION
**Pull request by NVnavkumar**
Fixes #8330.

Depends on https://github.com/NVIDIA/spark-rapids-jni/pull/1175 being merge first.

This completes #8161 and adds support for computing Remainder w/ Decimal 128 operands where higher precision is required than fits into the DECIMAL128 value. It calls the custom code from the spark-rapids-jni PR when the needed precision for the operands is greater than DECIMAL128_MAX_PRECISION.

### CODE REVIEW

**Comment by NVnavkumar** [comment]
build

**Comment by NVnavkumar** [comment]
Investigating Databricks compile failure. Looks unrelated to this branch.

**Comment by NVnavkumar** [comment]
> Investigating Databricks compile failure. Looks unrelated to this branch.

Looks like a new build failure with Databricks 11.3. Will file an issue.

**Comment by NVnavkumar** [comment]
Blocked until https://github.com/NVIDIA/spark-rapids/issues/8460 is resolved

**Comment by NVnavkumar** [comment]
build

**Comment by NVnavkumar** [comment]
build

**Comment by NVnavkumar** [comment]
build

File: integration_tests/src/main/python/arithmetic_ops_test.py [hunk 1]
```diff
@@ -398,17 +398,12 @@ def test_mod_mixed(lhs, rhs):
...
-@pytest.mark.parametrize('rhs', [DecimalGen(27,7), DecimalGen(30,10), DecimalGen(38,1)], ids=idfn)
-def test_mod_mixed_overflow_fallback(lhs, rhs):
-    assert_gpu_fallback_collect(
-        lambda spark : two_col_df(spark, lhs, rhs).selectExpr(f"a % b"), "Remainder")
+# @pytest.mark.skipif(not is_databricks113_or_later() and not is_spark_340_or_later(), reason="https://github.com/NVIDIA/spark-rapids/issues/8330")
+@pytest.mark.parametrize('lhs', [DecimalGen(38,0), DecimalGen(37,2), DecimalGen(38,5), DecimalGen(38,-10), DecimalGen(38,7)], ids=idfn)
+@pytest.mark.parametrize('rhs', [DecimalGen(27,7), DecimalGen(30,10), DecimalGen(38,1), DecimalGen(36,0), DecimalGen(28,-7)], ids=idfn)
+def test_mod_mixed_decimal128(lhs, rhs):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark : two_col_df(spark, lhs, rhs).selectExpr("a", "b", f"a % b"))
```
Comment (revans2) [inline integration_tests/src/main/python/arithmetic_ops_test.py:406]: nit: why the format string? `f"a % b"`???

**Review by revans2 — APPROVED** [review]

**Pull request merged**